plugin/backend_linux.go

```diff
@@ -340,12 +342,16 @@ func (pm *Manager) Pull(ctx context.Context, ref reference.Named, name string, m
 		return err
 	}
 
-	p, err := pm.createPlugin(name, dm.configDigest, dm.blobs, tmpRootFSDir, &privileges)
+	opts = append(opts, func(p *v2.Plugin) {
+		p.PluginObj.PluginReference = ref.String()
+	})
```

Comment:
It's unlikely to ever be a real problem, but I think this would have weird side effects if a subset of a slice was passed into `Pull`.

i.e.

```go
opts := []CreateOpt{opt1, opt2, opt3}
Pull(..., opts[:2]...)
// after the call, opts[2] is overwritten by the `append`.
```